### ISSUE TITLE
Input tag directly in properties

### ISSUE THREAD

**Issue by greetclammy** [Gorkycreator/obsidian-quick-tagger#15]
When executing the "Quick Tagger: Add tag" command, could it be possible to add an option to move cursor to the “tags” property in YAML frontmatter and input the tag there? Instead of inputting a tag via the modal. Similar to how the native Obsidian "Add file property" command behaves.

This would remove unnecessary friction and allow one:

1. To add multiple tags in a quick succession, separated only by pressing Enter instead of having to use a designated hotkey for each tag to add.
2. To see note and frontmatter content (including tags already in place) when inputting a tag.

Alternatively, if one has the "Properties view" core plugin enabled, the plugin could open the "File properties" tab in sidebar (or switch to it if it's open already).

So, ideally, there would be three options to choose from:
1. Modal
2. Frontmatter
3. File properties pane

Thank you for this great plugin and considering this request!

**Comment by Gorkycreator** [comment]
Thanks for the suggestion! I like this idea a lot, it fits in nicely with my goals for the plugin. Fleshing it out a bit:

- Having three options to pick with each tag entry would be cumbersome. Instead, there should be an option in the settings to add via `plugin modal` or `obsidian properties`. If the former is selected, the "Add tag" command uses the current method, if the latter, it moves the cursor or triggers the file properties input depending on the user's settings for file properties plugin.
- Bulk editing doesn't work with this, so right-click context menus and other entry points would still always default to the modal input. This would also preserve the advantage of not having to open a note to tag it.

I feel fairly confident I can get both the frontmatter cursor and the properties panel focused, but will need to do further investigation before confirming the viability.

Related to this pain point, I have a prototype modal to add multiple tags at once where it loops tag entry until a final command is selected, but I'm not quite happy with it yet:

![image](https://github.com/user-attachments/assets/f5cb1fb7-2ecf-4074-9e8c-cbb997742ef5)

**Comment by greetclammy** [comment]
>there should be an option in the settings to add via plugin modal or obsidian properties
Yep, that's what I meant 👍

>depending on the user's settings for file properties plugin
One (myself included) might want to have the option to insert tag to frontmatter available still, even if they have the *Properties view* plugin enabled. I think it's best if this was configured in Quick Tagger's settings.

>Bulk editing doesn't work with this
Hm, I don't follow. How can one bulk edit with Quick Tagger? And what do you mean by "this" here exactly?

**Comment by Gorkycreator** [comment]
> might want to have the option to insert tag to frontmatter available still, even if they have the Properties view plugin enabled.

Fair enough! Perhaps it could pop a warning if its configured for file properties, but the core plugin is not enabled.

> How can one bulk edit with Quick Tagger?

You can selecting multiple files and right-click on them to add or remove a tag. The search panel's menu also has an option to tag all search results. Adding a tag in the frontmatter or file properties doesn't make sense in this context, as those are inputs for a single note.
